purchases/src/main/kotlin/com/revenuecat/purchases/coroutines.kt

```diff
@@ -0,0 +1,27 @@
+package com.revenuecat.purchases
+
+import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+/**
+ * Get latest available purchaser info.
+ * Coroutine friendly version of [Purchases.getCustomerInfo].
+ *
+ * @return The [CustomerInfo] or a [PurchasesException] with the [PurchasesError]
+ */
+suspend fun Purchases.awaitCustomerInfo(): CustomerInfo {
```

Comment:
Also it's worth mentioning that leaving the same name may be confusing to clients because you don't if you're calling the sync or the async version of the function.

I suggested `await` in the first place because that's what Google uses in their KTX libraries (e.g. [`GoogleMap.awaitSnapshot`](https://github.com/googlemaps/android-maps-ktx/blob/31d4bdec1b30eb5cd5c72a385a725d05cd29e14a/maps-ktx/src/main/java/com/google/maps/android/ktx/GoogleMap.kt#LL202C1-L202C1)) but I also like `async` nomenclature 👍

Reply:
Yeah... I guess I like similar methods to be together in the autocomplete in AS but I can see there can be confusion when using the same name... Also, I was now thinking that Google uses the `async` suffix in non-coroutines methods in their billing library: https://developer.android.com/reference/com/android/billingclient/api/BillingClient#queryPurchasesAsync(com.android.billingclient.api.QueryPurchasesParams,%20com.android.billingclient.api.PurchasesResponseListener).

So I think it's ok to keep them as `await....`. 

> I guess we could not have these as extension functions and name them the same way as the regular functions. 

I'm a bit concerned of growing `Purchases` even more. I was thinking about separating them to a different class as either a subclass or a different class entirely but don't like those options... I think I prefer to keep them as extension functions. Lmk if you think otherwise though!

Reply:
I prefer `await` versus `async` because `await` is more of an indicative that it is a suspend function